common/flatpak-dir.c

```diff
@@ -2127,10 +2131,23 @@ flatpak_dir_pull_extra_data (FlatpakDir          *self,
 
       /* TODO: Download to disk to support resumed downloads on error */
 
-      ensure_soup_session (self);
-      bytes = flatpak_load_http_uri (self->soup_session, extra_data_uri, NULL, NULL,
-                                     extra_data_progress_report, &extra_data_progress,
-                                     cancellable, error);
+      base_dir = flatpak_get_user_base_dir_location ();
+      extra_local_dir = g_file_get_child (base_dir, "extra-data");
+      extra_local_file = g_file_get_child (g_file_get_child (extra_local_dir, extra_data_sha256), extra_data_name);
+      if (g_file_query_exists (extra_local_file, cancellable))
+        {
+          g_debug ("Loading extra-data from local file %s", g_file_get_path (extra_local_file));
+          extra_local_stream = (GInputStream *) g_file_read (extra_local_file, cancellable, error);
+          bytes = g_input_stream_read_bytes (extra_local_stream, download_size, cancellable, error);
```

Comment:
g_input_stream_read_bytes is just a more bindable version of g_input_stream_read(). It doesn't guarantee that you read the entire file, etc. Use g_file_load_contents instead and put the result (if the length is correct) in a GBytes.